### DISCUSSION TITLE
import { createStore } from 'zustand/vanilla' or from zustand ?

### DISCUSSION THREAD

**Discussion by vaynevayne** [pmndrs/zustand#1866] [Q&A]
I wanted to use the provider pattern, but found that there are two ways to write it

https://docs.pmnd.rs/zustand/guides/initialize-state-with-props#store-creator-with-createstore

```jsx
import { createStore } from 'zustand'
createStore<BearState>()()

```

https://github.com/pmndrs/zustand#using-zustand-without-react

```jsx
import { createStore } from 'zustand/vanilla'

const store = createStore(() => ({ ... }))
```

https://github.com/pmndrs/zustand#react-context
```jsx
import { createStore, useStore } from 'zustand'

const store = createStore(...) // vanilla store without hooks


```

It's a mess. Where do I get it from?

One more small () or two ()?

**Comment by dbritto-dev** [comment]
@vaynevayne currently there're two ways to create a store uncurried and curried version both works well on js when you need to use it w/o middlewares on typescript the story is different if you want to use middlewares you must use the curried version. BTW, you could always use the curried version and that would be fine
